src/monster.cpp

```diff
@@ -458,12 +452,124 @@ int monster::print_info(WINDOW* w, int vStart, int vLines, int column) const
 
 std::string monster::extended_description() const
 {
-    // @todo Add tons of info here
-    std::string ret = string_format( _ ( "This is a %s" ), name().c_str() );
-    ret += "\n--\n";
+    std::ostringstream ss;
+
+    nc_color color = c_white;
+    std::string attitude;
+    get_Attitude( color, attitude );
+    attitude = get_tag_from_color( color ) + attitude;
+
+    ss << string_format( _( "This is a %s. %s" ), name().c_str(), attitude.c_str() ) << std::endl;
+    if( !get_effect_status().empty() ) {
+        ss << string_format( _( "<stat>It is %s.</stat>" ), get_effect_status().c_str() ) << std::endl;
+    }
+
+    ss << "--" << std::endl;
     auto hp_bar = hp_description( hp, type->hp );
-    ret += get_tag_from_color( hp_bar.second ) + hp_bar.first;
-    return replace_colors( ret );
+    ss << get_tag_from_color( hp_bar.second ) << hp_bar.first << std::endl;
+
+    ss << "--" << std::endl;
+    ss << string_format( _( "<dark>%s</dark>" ), type->description.c_str() ) << std::endl;
+    ss << "--" << std::endl;
+
+    std::string size_name;
+    if( get_size() == MS_TINY ) {
+        size_name = _( "tiny" );
+    }
+    if( get_size() == MS_SMALL ) {
+        size_name = _( "small" );
+    }
+    if( get_size() == MS_MEDIUM ) {
+        size_name = _( "medium" );
+    }
+    if( get_size() == MS_LARGE ) {
+        size_name = _( "large" );
+    }
+    if( get_size() == MS_HUGE ) {
+        size_name = _( "huge" );
+    }
+    ss << string_format( _( "It is %s in size." ), size_name.c_str() ) << std::endl;
+
+    std::vector<std::string> types;
+    if( type->has_flag( MF_ANIMAL ) ) {
+        types.emplace_back( _( "animal") );
+    }
+    if( type->in_species( ZOMBIE ) ) {
+        types.emplace_back( _( "zombie" ) );
+    }
+    if( type->in_species( FUNGUS ) ) {
+        types.emplace_back( _( "fungus" ) );
+    }
+    if( type->in_species( INSECT ) ) {
+        types.emplace_back( _( "insect" ) );
+    }
+    if( type->in_species( ABERRATION ) ) {
+        types.emplace_back( _( "aberration" ) );
+    }
+    if( !types.empty() ) {
+        ss << string_format( _( "It is %s." ),
+                             enumerate_as_string( types ).c_str() ) << std::endl;
```

Comment:
Same as above.
Also wording: "It is zombie" sounds weird, "It is a zombie, an aberration, and an animal" would be better.